src/spec.c

```diff
@@ -866,6 +929,140 @@ static int parseVectorField_flat(FieldSpec *fs, VecSimParams *params, ArgsCursor
+        QERR_MKBADARGS_AC(status, VECSIM_ALGO_PARAM_MSG(VECSIM_ALGORITHM_SVS, VECSIM_WINDOW_SIZE), rc);
+        return 0;
+      }
+    } else if (AC_AdvanceIfMatch(ac, VECSIM_NUM_THREADS)) {
+      if ((rc = AC_GetSize(ac, &params->algoParams.svsParams.num_threads, AC_F_GE1)) != AC_OK) {
+        QERR_MKBADARGS_AC(status, VECSIM_ALGO_PARAM_MSG(VECSIM_ALGORITHM_SVS, VECSIM_NUM_THREADS), rc);
+        return 0;
+      } else if (params->algoParams.svsParams.num_threads > MAX_WORKER_THREADS) {
+           QueryError_SetWithUserDataFmt(status, QUERY_EPARSEARGS, "NUM_THREADS value exceeds MAX_WORKER_THREADS. ", "Not more than %d is allowed", MAX_WORKER_THREADS);
+          return 0;
+      }
+    } else if (AC_AdvanceIfMatch(ac, VECSIM_QUANT_BITS)) {
+      if ((rc = parseVectorField_GetQuantBits(ac, &params->algoParams.svsParams.quantBits)) != AC_OK) {
+        QERR_MKBADARGS_AC(status, VECSIM_ALGO_PARAM_MSG(VECSIM_ALGORITHM_SVS, VECSIM_QUANT_BITS), rc);
+        return 0;
+      }
+    }
+    else if (AC_AdvanceIfMatch(ac, VECSIM_WSSEARCH)) {
+      if ((rc = AC_GetSize(ac, &params->algoParams.svsParams.search_window_size, AC_F_GE1)) != AC_OK) {
+        QERR_MKBADARGS_AC(status, VECSIM_ALGO_PARAM_MSG(VECSIM_ALGORITHM_SVS, VECSIM_WSSEARCH), rc);
+        return 0;
+      }
+    } else if (AC_AdvanceIfMatch(ac, VECSIM_MAX_CANDIDATE_POOL_SIZE)) {
+      if ((rc = AC_GetSize(ac, &params->algoParams.svsParams.max_candidate_pool_size, AC_F_GE1)) != AC_OK) {
+        QERR_MKBADARGS_AC(status, VECSIM_ALGO_PARAM_MSG(VECSIM_ALGORITHM_SVS, VECSIM_MAX_CANDIDATE_POOL_SIZE), rc);
+        return 0;
+      }
+    } else if (AC_AdvanceIfMatch(ac, VECSIM_PRUNE_TO)) {
+      if ((rc = AC_GetSize(ac, &params->algoParams.svsParams.prune_to, AC_F_GE1)) != AC_OK) {
+        QERR_MKBADARGS_AC(status, VECSIM_ALGO_PARAM_MSG(VECSIM_ALGORITHM_SVS, VECSIM_PRUNE_TO), rc);
+        return 0;
+      }
+    } else if (AC_AdvanceIfMatch(ac, VECSIM_ALPHA)) {
+      double tmpd;
+      if ((rc = AC_GetDouble(ac, &tmpd, AC_F_GE0)) != AC_OK) {
+        QERR_MKBADARGS_AC(status, VECSIM_ALGO_PARAM_MSG(VECSIM_ALGORITHM_SVS, VECSIM_ALPHA), rc);
+        return 0;
+      }
+      params->algoParams.svsParams.alpha = (float)tmpd;
+    } else if (AC_AdvanceIfMatch(ac, VECSIM_USE_SEARCH_HISTORY)) {
+      if ((rc = parseVectorField_GetOption(ac, &params->algoParams.svsParams.use_search_history)) != AC_OK) {
+        QERR_MKBADARGS_AC(status, VECSIM_ALGO_PARAM_MSG(VECSIM_ALGORITHM_SVS, VECSIM_USE_SEARCH_HISTORY), rc);
+        return 0;
+      }
+    } else if (AC_AdvanceIfMatch(ac, VECSIM_EPSILON)) {
+      if ((rc = AC_GetDouble(ac, &params->algoParams.svsParams.epsilon, AC_F_GE0)) != AC_OK) {
+        QERR_MKBADARGS_AC(status, VECSIM_ALGO_PARAM_MSG(VECSIM_ALGORITHM_HNSW, VECSIM_EPSILON), rc);
+        return 0;
+      }
+    } else if (AC_AdvanceIfMatch(ac, VECSIM_TRAINING_THRESHOLD)) {
```

Comment:
This argument is valid only if `params->algoParams.svsParams.quantBits != 0` (i.e., if some compression algorithm is requested), and it should never be lower than `DEFAULT_BLOCK_SIZE`